### ISSUE TITLE
nil pointer dereference on arm7 v0.31.2 release

### ISSUE THREAD

**Issue by gurre** [Jeffail/benthos#102]
Using release v0.31.2.

```shell
$ cat /proc/cpuinfo
processor	: 0
model name	: ARMv7 Processor rev 5 (v7l)
BogoMIPS	: 38.40
Features	: half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm 
CPU implementer	: 0x41
CPU architecture: 7
CPU variant	: 0x0
CPU part	: 0xc07
CPU revision	: 5

# ...

Hardware	: BCM2835
Revision	: a01041
Serial		: -
```

```shell
$ cat benthos.yaml 
input:
  type: stdin
output:
  type: stdout

$ benthos -c benthos.yaml
{"@timestamp":"2018-09-16T14:09:43Z","level":"INFO","@service":"benthos","message":"Launching a benthos instance, use CTRL+C to close."}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x4 pc=0x12054]

goroutine 5 [running]:
github.com/Jeffail/benthos/lib/util/throttle.(*Type).Reset(0x1d19ce0)
	/home/ash/src/go/benthos/lib/util/throttle/type.go:141 +0x34
github.com/Jeffail/benthos/lib/input.(*Reader).loop(0x1ee9840)
	/home/ash/src/go/benthos/lib/input/reader.go:136 +0xbb4
created by github.com/Jeffail/benthos/lib/input.NewReader
	/home/ash/src/go/benthos/lib/input/reader.go:76 +0x208
```

**Comment by gurre** [comment]
It's the same for the v0.31.1 release

```shell
$ benthos
{"@timestamp":"2018-09-16T14:15:06Z","level":"INFO","@service":"benthos","message":"Launching a benthos instance, use CTRL+C to close."}
{"@timestamp":"2018-09-16T14:15:06Z","level":"INFO","@service":"benthos","message":"Listening for HTTP requests at: http://0.0.0.0:4195\n"}
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x4 pc=0x12054]

goroutine 5 [running]:
github.com/Jeffail/benthos/lib/util/throttle.(*Type).Reset(0x2919c80)
	/home/ash/src/go/benthos/lib/util/throttle/type.go:141 +0x34
github.com/Jeffail/benthos/lib/input.(*Reader).loop(0x2ad9540)
	/home/ash/src/go/benthos/lib/input/reader.go:136 +0xbb4
created by github.com/Jeffail/benthos/lib/input.NewReader
	/home/ash/src/go/benthos/lib/input/reader.go:76 +0x208
```

**Comment by Jeffail** [comment]
Hey @gurre, thanks for raising this. It looks as though this could be a problem in multiple areas of Benthos where 64-bit integers are atomically modified. I believe I've fixed this issue as well as a few others but there's a possibility that there might be other nasties lurking in there.

If you see any more crashes let me know and I'll get them fixed ASAP, it's fairly easy to fix once the issue is exposed. Either reopen this ticket or start a new one, either is fine.

Fixed: https://github.com/Jeffail/benthos/commit/56413a4865a4c243c7762ea28392837d0dc00bc7
Released: https://github.com/Jeffail/benthos/releases/tag/v0.31.3
